### DISCUSSION TITLE
How to hard reset the form, so multiple submits are possible?

### DISCUSSION THREAD

**Discussion by enricoschaaf** [react-hook-form/react-hook-form#2663] [Q&A] [answered]
I want to allow a form to be submitted as often as the person wants, but when I call the reset function and want to submit the form again the reValidateMode is active and the form gets validated on change, which is not what I want.

**Comment by bluebill1049** [comment, marked as answer]
codesandbox?

**Reply by enricoschaaf** [reply]
https://codesandbox.io/s/vigilant-lamarr-h1jk8?file=/src/App.js, when I fill it out the second time it gets validated onChange not onBlur, I would like that it behaves like the first time it submits when its successfully submitted.😊

**Reply by bluebill1049** [reply]
not if you had a succesfully submit: https://codesandbox.io/s/blue-glitter-o1vrx

**Reply by enricoschaaf** [reply]
I have to say I dont understand your example, sure if the only validation is required, then there will be no validation onChange because with one caracter entered there is no more error. I have a short video of what I mean.
![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/54645197/90837020-186eeb80-e351-11ea-8e0b-6d1ab40a31e2.gif)

**Reply by bluebill1049** [reply]
Take a look here: https://codesandbox.io/s/blue-glitter-o1vrx?file=/src/App.js

**Reply by enricoschaaf** [reply]
Works like a charm, thanks you very much! 😊

**Reply by TroyWolf** [reply]
This gave me some new ideas, but the issue with @bluebill1049 codesandbox is the form reset happens even on a non-successful submit. That is, if you have 3 fields in your form and all are required and you click submit after only one has been completed, the hook will fire and reset() will reset the form when what you really want is to show the validation errors and let the user continue.

I modified to do this:
```
  // I want to completely reset this form on successful submission
  useEffect(() => {
    if (isValid && isSubmitted) {
      reset()
    }
  }, [isSubmitted, isValid, reset])
```
This seemed to work at first, but then I realized my form would reset on the first keyUp that caused the form to become valid. Oops. It also throws these console warnings:
```
`formState.isValid` is applicable with `onChange` or `onBlur` mode. https://react-hook-form.com/api#formState
```

**Reply by TroyWolf** [reply]
Others have argued whether `isSubmitted` should only be true if the validation passes. I am in that camp, but also not sure of all the ways RHF uses that boolean.

**Reply by TroyWolf** [reply]
I was able to have a form completely reset to brand-new state by doing this upon successful submit:

HACK ALERT! 
`window.setTimeout(reset, 250)`

Without the timeout, the reset fires before all the other formState stuff is set internally by RHF. My fields would be reset, but isSubmitted was still `true`, which caused the validation to fire on keyUp instead of on submit like a fresh form. Or maybe it's an issue with `submitCount`? Either way, it's a timing issue.

Using `setTimeout` is weak! I'd much rather respond to an actual event that can be trusted.
